### ISSUE TITLE
Terminal shuts down upon browser restart

### ISSUE THREAD

**Issue by markusweimer** [MicrosoftDocs/vsonline#239]
When I close the browser tab with VSO inside and the reconnect, the *editor* maintains a remarkable amount of state, down to the cursor position. That is awesome!

However, the same does not apply to programs running in the terminals. For example, a running `ipython` session gets restarted when I reconnect from a new browser tab.

I can work around this using `screen` or such, but it would be awesome to not have to do that. Are there any plans to support persistence of the Terminal between browser changes?

**Comment by nikmd23** [comment]
Hi @markusweimer! Question for you, how long between the tab close and reopen are we talking about here? Longer than your environment's auto suspend policy?

**Comment by markusweimer** [comment]
> how long between the tab close and reopen are we talking about here? Longer than your environment's auto suspend policy?

No, it happens right away, even when opening the new browser tab in a few seconds.

**Comment by lostintangent** [comment]
Unfortunately, this is our current behavior. We started with enabling "editor state" to be retained across connections, and moving forward, we plan to address runtime state as well. That said, this is a fairly large work item, and so it's TBD when that will come online. 

To @nikmd23's question, is it safe to assume that you only expect the terminals/processes to be retained as long as the environment hasn't been suspended yet? (e.g. you refresh the browser/disconnect and connect from another machine).

**Comment by markusweimer** [comment]
>  is it safe to assume that you only expect the terminals/processes to be retained as long as the environment hasn't been suspended yet? (e.g. you refresh the browser/disconnect and connect from another machine).

That would be great. I have no idea how you'd do it beyond the shutdown time anyway. Maybe freeze the processes?
